### PR TITLE
feat: expose CometSchemaImporter in BatchReader

### PR DESCRIPTION
**Pull request by huaxingao**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Add a getter for `CometSchemaImporter importer` in `BatchReader` for iceberg integration. 
In iceberg, we will do something like this:

```
          CometSchemaImporter importer = batchReader.getImporter();
          if (importer != null) {
            importer.close();
          }
          importer = new CometSchemaImporter(BatchReader.ALLOCATOR);

          AbstractColumnReader delegate = 
             Utils.getColumnReader(sparkType, descriptor, importer, batchSize, useDecimal128, useLazyMaterialization);
```

### CODE REVIEW

**Comment by huaxingao** [comment]
cc @viirya

**Comment by viirya** [comment]
> CometSchemaImporter importer = batchReader.getImporter();
          if (importer != null) {
            importer.close();
          }
          importer = new CometSchemaImporter(BatchReader.ALLOCATOR);

Hmm, by doing this, you close the internal importer in batch reader, but open a new importer outside it?

File: common/src/main/java/org/apache/comet/parquet/BatchReader.java [hunk 1]
```diff
@@ -231,6 +231,10 @@ public BatchReader(AbstractColumnReader[] columnReaders) {
     this.taskContext = TaskContext$.MODULE$.get();
   }
 
+  public CometSchemaImporter getImporter() {
+    return importer;
+  }
```
Comment (viirya) [inline common/src/main/java/org/apache/comet/parquet/BatchReader.java:236]: If you will create `CometSchemaImporter` in Iceberg to use, it looks like we don't need to get this importer.

Comment (viirya) [reply common/src/main/java/org/apache/comet/parquet/BatchReader.java:236]: This PR looks redundant.

Comment (huaxingao) [reply common/src/main/java/org/apache/comet/parquet/BatchReader.java:236]: OK. I will create a CometSchemaImporter in iceberg. Will close this PR.

**Pull request closed without merging**